src/node_buffer.cc

```diff
@@ -219,7 +219,10 @@ MaybeLocal<Object> New(Isolate* isolate,
   CHECK(actual <= length);
 
   if (actual < length) {
-    data = static_cast<char*>(realloc(data, actual));
+    char* new_data = static_cast<char*>(realloc(data, actual));
```

Comment:
If realloc fails, the whole program just asserts. It wouldn't harm to free the old data before that, but there is no actual leak in apps here.